flytekit/sdk/tasks.py

```diff
@@ -35,7 +36,7 @@ def my_task(wf_params, in1, in2, out1, out2):
     """
 
     def apply_inputs_wrapper(task):
-        if not isinstance(task, _sdk_runnable_tasks.SdkRunnableTask):
+        if not isinstance(task, _sdk_runnable_tasks.SdkRunnableTask) and not isinstance(task, presto_task.SdkPrestoTask):
```

Comment:
this logic to prevent additional application of inputs to a presto task should be contained within the presto_task object. Consider overriding the `add_inputs` method in presto task

Reply:
OK I see a little bit more of what you are trying to do here now. I think the best way forward depends on whether you stick with passing `inputs(...)` or go to `{'in1': Types.String}` style when calling the constructor. But we really shouldn't stack isinstance checks where we can avoid it (they really should just exist for robust error messaging--not specifically for uncontained logic)